### ISSUE TITLE
Design for dashboards.

### ISSUE THREAD

**Issue by ryzhyk** [feldera/feldera#2288]

**Comment by lalithsuresh** [comment]
> WebConsole integration:
>    The Webconsole should at least contain a link to open the dashboard for the current pipeline. I don't know if it makes sense to embed the dashboard as a tab in the WebConsole

Probably not. Easier to give them a link to our documentation to do fancier observability.

> How do we ship Prometheus + Grafana is the k8s form factor, which will require multitenancy support. We expect enterprise users to hook Feldera into their existing Prometheus/Grafana deployments for ops, but it would still be useful to offer builtin dashboard for developers.

The good thing about using the pipeline-manager and service discovery approach is that Grafana/Prometheus it can be completely external. The developer could just as well run docker compose on their laptop for Grafana/Prometheus and aim them at the pipeline-manager, even if the latter is running somewhere else like our sandbox.

**Comment by ryzhyk** [comment]
> > WebConsole integration:
> > The Webconsole should at least contain a link to open the dashboard for the current pipeline. I don't know if it makes sense to embed the dashboard as a tab in the WebConsole
> 
> Probably not. Easier to give them a link to our documentation to do fancier observability.

Fair. But a link inside the UI to open Grafana in a browser tab should be trivial and can be a big QoL thing.

> > How do we ship Prometheus + Grafana is the k8s form factor, which will require multitenancy support. We expect enterprise users to hook Feldera into their existing Prometheus/Grafana deployments for ops, but it would still be useful to offer builtin dashboard for developers.
> 
> The good thing about using the pipeline-manager and service discovery approach is that Grafana/Prometheus can be completely external. The developer could just as well run docker compose on their laptop for Grafana/Prometheus and aim them at the pipeline-manager, even if the latter is running somewhere else like our sandbox.

Good point.

**Comment by gz** [comment]
> Each pipeline exports Prometheus metrics via the /metrics endpoint.

do we still keep the current localhost:8081/metrics endpoint to monitor all pipelines at the same time? Also the metrics should be the same except that :8081 has metrics for all `n` pipelines that are running. I don't know how the prometheus discovery works, but ideally I  like to have one dashboard where all pipelines just show up that I never have to modify etc.

**Comment by lalithsuresh** [comment]
> do we still keep the current localhost:8081/metrics endpoint to monitor all pipelines at the same time? Also the metrics should be the same except that :8081 has metrics for all n pipelines that are running. I don't know how the prometheus discovery works, but ideally I like to have one dashboard where all pipelines just show up that I never have to modify etc.

I didn't realize we had this, but this should definitely not be exposed in a setting with auth.

**Comment by gz** [comment]
this is on 8081 its not exposed

**Comment by ryzhyk** [comment]
From memory, It doesn't matter where Prometheus scrapes the metrics from, they can all be used in the same dashboard.

**Comment by gz** [comment]
I think the annoying part would be if you had to configure a new endpoint just because you added aa pipeline. I dont know if thats needed or not.

**Comment by ryzhyk** [comment]
Definitely not, that's what the service discovery part supposed to solve.

**Comment by lalithsuresh** [comment]
Yup. The service discovery part will solve this, and it keeps it multi-tenancy aware.
